### ISSUE TITLE
Controller runtime requires setting a logger

### ISSUE THREAD

**Issue by sagikazarmark** [kubernetes-sigs/e2e-framework#270]
When using the resources package with a decoder, I receive the following error:

```
[controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:
goroutine 1 [running]:
runtime/debug.Stack()
        /nix/store/bvihxp3hnnvpjilgqw1srfqm4dyx6xa5-go-1.20.4/share/go/src/runtime/debug/stack.go:24 +0x6c
sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/log.go:59 +0xc4
sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).WithName(0xc0003f1bc0, {0x10458d0cf, 0x14})
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/log/deleg.go:147 +0x38
github.com/go-logr/logr.Logger.WithName({{0x104ba90f0, 0xc0003f1bc0}, 0x0}, {0x10458d0cf, 0x14})
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/github.com/go-logr/logr@v1.2.4/logr.go:336 +0x50
sigs.k8s.io/controller-runtime/pkg/client.newClient(0xc00026a900, {0x0, 0xc000199650, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/client/client.go:115 +0xa0
sigs.k8s.io/controller-runtime/pkg/client.New(0xc000000000?, {0x0, 0xc000199650, {0x0, 0x0}, 0x0, {0x0, 0x0}, 0x0})
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/client/client.go:101 +0x90
sigs.k8s.io/e2e-framework/klient/k8s/resources.New(0xc00026a900)
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.2.0/klient/k8s/resources/resources.go:61 +0xc8
github.com/bank-vaults/vault-secrets-webhook/e2e.installVault({0x104ba73d8, 0xc000120a50}, 0xc000407e80)
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/e2e/main_test.go:140 +0x4c
sigs.k8s.io/e2e-framework/pkg/env.(*action).run(0xc000153c28, {0x104ba73d8, 0xc000120a50}, 0xc000407e80)
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.2.0/pkg/env/action.go:135 +0x16c
sigs.k8s.io/e2e-framework/pkg/env.(*testEnv).Run(0xc000488a80, 0xc000463380?)
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/.devenv/state/go/pkg/mod/sigs.k8s.io/e2e-framework@v0.2.0/pkg/env/env.go:369 +0x300
github.com/bank-vaults/vault-secrets-webhook/e2e.(*reverseFinishEnvironment).Run(0xc000463290, 0xc00041ee38?)
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/e2e/main_test.go:245 +0xa0
github.com/bank-vaults/vault-secrets-webhook/e2e.TestMain(0x0?)
        /Users/marksk/Projects/bank-vaults/vault-secrets-webhook/e2e/main_test.go:93 +0xd98
main.main()
        _testmain.go:49 +0x308
```

Based on the stack, the problem is caused by the following code:

```
	r, err := resources.New(cfg.Client().RESTConfig())
	if err != nil {
		return ctx, err
	}
```

e2e-framework version: 0.2.0

Adding the following piece of code fixes the problem:

```go
package e2e

import (
	// ...

	"k8s.io/klog/v2"
	"sigs.k8s.io/controller-runtime/pkg/log"
	// ...
)

var testenv env.Environment

func TestMain(m *testing.M) {
	testenv = env.New()

	log.SetLogger(klog.NewKlogr())

	// ...
}
```

**Comment by harshanarayana** [comment]
@sagikazarmark  Do you happen to have an example code that you can share here to reproduce the problem? It will also be useful to add that to the UTs we have to prevent issues like this in the future

**Comment by sagikazarmark** [comment]
@harshanarayana I gave you the offending line based on the stack trace above. If you'd like to see it in action:

1. Check out the following repository (at the linked commit): https://github.com/bank-vaults/vault-secrets-webhook/blob/d335c5e56358d90011f40d8566d21c5057d10fb7/e2e/main_test.go
2. Remove lines 32 and 59 (`log.SetLogger(klog.NewKlogr())` and the relevant import)
3. Run `LOG_VERBOSE=true make test-e2e-local`

**Comment by harshanarayana** [comment]
@sagikazarmark #276 should have fixed the issue you are running into.. Please check and confirm once and we can close this

**Comment by harshanarayana** [comment]
/close 

Fixed in #276
